language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/SelectClauseNodeContext.java

```diff
@@ -61,12 +65,37 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
         } else {
             completionItems.addAll(this.expressionCompletions(context));
             completionItems.add(new SnippetCompletionItem(context, Snippet.CLAUSE_ON_CONFLICT.get()));
+            
+            if (containsGroupByNode(node)) {
```

Comment:
Consider the following case. 
![image](https://github.com/ballerina-platform/ballerina-lang/assets/35211477/0a1c2e49-e421-421a-8ca7-84ab0580f68f)

it syas that  'join() is not defined. Same for the avg(), etc. We have to revisit and see what are the set of lang libs that are allowed.